internal/builder/packages.go

```diff
@@ -138,9 +152,18 @@ func FindBuildPackagesDirectory() (string, bool, error) {
 
 // BuildPackage function builds the package.
 func BuildPackage(options BuildOptions) (string, error) {
-	destinationDir, err := BuildPackagesDirectory(options.PackageRoot)
-	if err != nil {
-		return "", errors.Wrap(err, "can't locate build directory")
+	var err error
+	var destinationDir string
+	if options.buildDir == "" {
+		destinationDir, err = BuildPackagesDirectory(options.PackageRoot)
+		if err != nil {
+			return "", errors.Wrap(err, "can't locate build directory")
+		}
+	} else {
+		destinationDir, err = buildPackagesDirectory(options.buildDir, options.PackageRoot)
+		if err != nil {
+			return "", errors.Wrap(err, "can't locate build directory")
+		}
```

Comment:
Why exactly do you need this if-else block? It looks like it complicated the current logic.

Reply:
I added `options.buildDir` to have a predictable place where packages are built for testing.

I can move this logic to the new internal method. I didn't modify the behaviour or interface of `BuildPackagesDirectory` because it is a exposed method, though it is in an internal package.

Reply:
Moved logic to the current `BuildPackagesDirectory` and changed the other use we have of this.